numpy/core/__init__.py

```diff
@@ -16,10 +16,31 @@
     from . import multiarray
 except ImportError as exc:
     msg = """
+
+IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!
+
 Importing the multiarray numpy extension module failed.  Most
 likely you are trying to import a failed build of numpy.
-If you're working with a numpy git repo, try `git clean -xdf` (removes all
-files not under version control).  Otherwise reinstall numpy.
+Here is how to proceed:
+- If you're working with a numpy git repository, try `git clean -xdf`
+  (removes all files not under version control) and rebuild numpy.
+- If you are simply trying to use the numpy version that you have installed:
+  your installation is broken - please reinstall numpy.
+- If you have already reinstalled and that did not fix the problem, then:
+  1. Check that you are using the Python you expect (e.g. check the path with
+     `import os; print(os.__file__)`), and that you have no directories in
```

Comment:
`print(sys.executable)` might be more valuable

Reply:
good point, will change

Reply:
Is there a good reason not to run this and print out the output in this message?

Reply:
good suggestion, done now